src/backend/distributed/utils/node_metadata.c

```diff
@@ -84,6 +92,17 @@ PG_FUNCTION_INFO_V1(master_initialize_node_metadata);
 PG_FUNCTION_INFO_V1(get_shard_id_for_distribution_column);
 
 
+static NodeMetadata
+DefaultNodeMetadata()
+{
+	NodeMetadata nodeMetadata = {
+		.nodeRole = InvalidOid,
+		.nodeRack = WORKER_DEFAULT_RACK,
```

Comment:
can we fill in the rest here?

Reply:
The rest is all set to 0/NULL by default when initializing like this: https://stackoverflow.com/a/330867/2570866

Do you want me to make those 0/NULL cases explicit?

Reply:
Then .nodeRole can be left out.